review CRUD 기능 구현 및 test code 작성

MyungHwanKim opened the pull request:
# Summary
feat : review CRUD 기능 구현

1. review 작성
- reservation에 boolean으로 isReview(예약 건에 대한 리뷰 작성 유무) 컬럼 추가
- isReview가 true일 경우 review 재작성 방지

2. review 목록/수정/삭제

test :  review CRUD 기능에 대한 test code 작성

1. 리뷰 작성
- 게스트가 존재하지 않는 경우
- 방이 존재하지 않는 경우
- 예약 내역이 존재하지 않는 경우
- 이미 예약에 대한 리뷰를 작성한 경우

2. 리뷰 확인
- 잘못된 접근일 경우

3. 리뷰 수정
- 게스트가 존재하지 않는 경우
- 리뷰가 존재하지 않는 경우
- 리뷰 카테고리가 존재하지 않는 경우

4. 리뷰 삭제
- 리뷰가 존재하지 않는 경우

# Relevant

- 커밋 ID : 59fc86d, a486bc9

# Screenshot
  관련된 스크린샷

# PR CheckList
아래의 요구사항을 충족하셨는지 확인하여 체크해주세요.

체크 방법은 [ ] 괄호 안에 x를 기입하시면 됩니다.

- [x] 작업한 내용에 대해 테스트를 진행했음.
- [x] 칸반보드에 해당 작업이 업데이트 되어있음.
- [x] commit 전, `git status`, `git pull`을 통해 반영되지 않은 코드가 없음을 확인함.

# To Reviewers
- reservation에 boolean으로 isReview(예약 건에 대한 리뷰 작성 유무) 컬럼 추가
- 이번 주 안으로 개발 최대한 마무리되도록 열심히 해요~~~!!

JmKanmo commented:
오... PR 템플릿 좋네요 ㅎㅎ  추가할 내용이라고 하면 .. 연관 이슈의 커밋 id 정도..? 일 것 같아요.   아 그리고 브랜치명 오타가 나셨는지 feat로 되잇으시네요. feature 풀네임으로 브랜치명 통일하면 좋을 것 같아용

Haze-S commented:
> 오... PR 템플릿 좋네요 ㅎㅎ 추가할 내용이라고 하면 .. 연관 이슈의 커밋 id 정도..? 일 것 같아요. 아 그리고 브랜치명 오타가 나셨는지 feat로 되잇으시네요. feature 풀네임으로 브랜치명 통일하면 좋을 것 같아용

넵! 오늘 저녁 퇴근후에 추가하도록 하겠습니다!
브랜치명은 줄인거였는데 다음 브랜치 생성부터는 통일하겠습니다ㅎㅎ 의견 감사합니다<3

BTSBRINGMEHERE commented:
PR 처음 사용해봐서 템플릿 좋네요!

MyungHwanKim commented:
저도 준모님이 얘기한 연관 이슈의 커밋 id 정도만 추가하면 괜찮을 거 같습니다.

나머지는 다 좋은 거 같습니다!!👍👍

MyungHwanKim committed:
feat : review CRUD 기능 구현

1. review 작성
- reservation에 boolean으로 isReview(예약 건에 대한 리뷰 작성 유무) 컬럼 추가
- isReview가 true일 경우 review 재작성 방지

2.review 목록/수정/삭제

MyungHwanKim committed:
test :  review CRUD 기능에 대한 test code 작성

1. 리뷰 작성
- 게스트가 존재하지 않는 경우
- 방이 존재하지 않는 경우
- 예약 내역이 존재하지 않는 경우
- 이미 예약에 대한 리뷰를 작성한 경우

2. 리뷰 확인
- 잘못된 접근일 경우

3. 리뷰 수정
- 게스트가 존재하지 않는 경우
- 리뷰가 존재하지 않는 경우
- 리뷰 카테고리가 존재하지 않는 경우

4. 리뷰 삭제
- 리뷰가 존재하지 않는 경우

JmKanmo commented:
코드 작성하시느라 수고 많으셨습니다. 

리뷰 form 클래스의 경우, 각각의 필드에  validation check를 추가해주시면 좋을 것 같습니다. 
저마다 코드 작성하는 방식이 다 잇어서 모두 통일하기는 쉽지 않은것같은데 ㅠ 
컨트롤러에  /register  API 클래스의 경우  넘어오는 form 데이터에 validation이 안맞는 경우 매개변수에 BindingResult bindingResult 추가해주고 아래와 같이 걸러서 예외로 반환해주면 더 좋지 않을까싶네용 ㅎㅎ  

```
    if (bindingResult.hasErrors()) {
            throw new ValidationException(ExceptionMessage.NOT_VALID_INPUT);
        }
```

Haze-S reviewed (approved):
고생하셨습니다 😊

MyungHwanKim committed:
review controller form의 validation check 및 form에 NotEmpty 적용
